libkbfs/folder_branch_ops.go

```diff
@@ -3694,21 +3697,79 @@ func (fbo *folderBranchOps) syncAllLocked(
 		return err
 	}
 
+	bps := newBlockPutState(0)
+	resolvedPaths := make(map[BlockPointer]path)
+	lbc := make(localBcache)
+
 	var cleanups []func(context.Context, *lockState, error)
 	defer func() {
 		for _, cf := range cleanups {
 			cf(ctx, lState, err)
 		}
 	}()
 
-	fbo.log.CDebugf(ctx, "Syncing %d file(s)", len(dirtyRefs))
-	bps := newBlockPutState(0)
-	resolvedPaths := make(map[BlockPointer]path)
+	// First prep all the directories.
+	fbo.log.CDebugf(ctx, "Syncing %d dirs(s)", len(dirtyDirs))
+	for _, ref := range dirtyDirs {
+		node := fbo.nodeCache.Get(ref)
+		if node == nil {
+			continue
+		}
+
+		dir := fbo.nodeCache.PathFromNode(node)
+		dblock, err := fbo.blocks.GetDirtyDir(ctx, lState, md, dir)
+		if err != nil {
+			return err
+		}
+
+		lbc[dir.tailPointer()] = dblock
```

Comment:
No, the new pointers aren't calculated until the `prepUpdateForPaths` later on.  Does something seem wrong here in that case?